packages/test/test-end-to-end-tests/src/test/container.spec.ts

```diff
@@ -744,13 +749,56 @@ describeCompat("Container", "NoCompat", (getTestObjectProvider) => {
 				() => runtimeDispose++,
 			);
 
-			container.close(new DataCorruptionError("expected", {}));
-			container.dispose(new DataCorruptionError("expected", {}));
+			container.close(new DataCorruptionError("close error", {}));
+			container.dispose(new DataCorruptionError("dispose error", {}));
 			assert.strictEqual(containerDisposed, 1, "Container should send disposed event");
 			assert.strictEqual(containerClosed, 1, "Container should send closed event");
 			assert.strictEqual(deltaManagerDisposed, 1, "DeltaManager should send disposed event");
 			assert.strictEqual(deltaManagerClosed, 1, "DeltaManager should send closed event");
 			assert.strictEqual(runtimeDispose, 1, "IContainerRuntime should send dispose event");
+			assert.strictEqual(
+				container.closedWithError?.message,
+				"close error",
+				"Expected the error that closed the container (not the dispose one)",
+			);
+		},
+	);
+
+	itExpects(
```

Comment:
nit. end to end test are costly to run and maintain, did you consider a unit test, or a local-server test?

Reply:
Great point.  I started by trying to unit test... there are NO existing unit tests for Container class! So decided not to be the tip of that spear. Local-server only is a good idea though.  Probably most of this file should be local-server only.

Reply:
Turns out all these e2e tests only run against local, so that's good.  We could move them to local-server-tests though at some point